pkg/controllers/user/alert/configsyncer/configsyncer.go

```diff
@@ -594,6 +647,13 @@ func (d *ConfigSyncer) addRecipients(notifiers []*v3.Notifier, receiver *alertco
 		}
 	}
 
+	if len(syncReceivers) > 0 {
+		err := d.syncReceiver(syncReceivers)
```

Comment:
1. It looks irrelevant to the purpose of `addRecipients()` which is to fill `alertconfig.Receiver` configurations.
2. It can be called multiple times given the way `addRecipients()` is invoked.

I think a proper place to do syncReceiver is in ConfigSyncer.sync() after alertmanager configs are handled.
Then get the latest receiver configs, update it only if it is changed.